src/components/adminArea/Drawer/UserEditForm.tsx

```diff
@@ -45,14 +74,29 @@ const UserEditForm = ({ currentUser, setIsLoading }) => {
         try {
             setIsLoading(true);
             const sendBadges = badges.map((badge) => badge._id);
+            const selectedSlugs = selectedNamespaces.map(ns => ns.slug);
+            const updatedRole = { ...role };
+
+            Object.keys(updatedRole).forEach((key) => {
```

Comment:
Nitpick: Consider using "role" instead of "key" for better clarity and semantic meaning